src/qt/treectrl.cpp

```diff
@@ -348,33 +349,46 @@ class wxQTreeWidget : public wxQtEventSignalHandler<QTreeWidget, wxTreeCtrl>
     {
         wxTreeCtrl* treeCtrl = GetHandler();
 
+        // Makes wxTreeCtrl::GetSelection() return previous value.
+        {
+            wxQtEnsureSignalsBlocked blocker(this);
+            setCurrentItem(previous);
+        }
+
         wxTreeEvent changingEvent(
             wxEVT_TREE_SEL_CHANGING,
             treeCtrl,
             wxQtConvertTreeItem(current)
         );
-
         changingEvent.SetOldItem(wxQtConvertTreeItem(previous));
-        EmitEvent(changingEvent);
 
-        if ( !changingEvent.IsAllowed() )
+        if (EmitEvent(changingEvent) && !changingEvent.IsAllowed())
         {
-            wxQtEnsureSignalsBlocked blocker(this);
-            setCurrentItem(previous);
+            // Has to be deferred, otherwise it doesn't have any effect.
+            QTimer::singleShot(0,
```

Comment:
Is this better than using `CallAfter()`?

Reply:
This is no longer needed with AliKet's patch.